Basic command line arg parsing added

sebhz opened the pull request:
argc/argv fully handled in their own module. Rest of the code mostly unchanged. Basic tests added for the new arg parsing module.
For now this only displays a very basic help as well as the version name.

If needed be we can reuse the argparse file to overwrite some of the options in cap32.cfg, but this would require reordering cap32.cfg parsing. This is probably overkill - except maybe to modify the path to cap32.cfg, for test and debug purpose.

ColinPitrat commented on test/argparse.cpp:
```diff
@@ -0,0 +1,38 @@
+#include "cap32.h"
+#include "argparse.h"
+
+TEST(ArgParseTest, parseArgsNoArg)
```
Not mandatory:
It would be cool to also be able to tests flags (especially if we plan to add more) but I recon this would require some over-engineering (what I can think of is a ArgParser class with a virtual Exit method and injectable streams in replacement of cout/cerr).

sebhz commented on src/argparse.cpp:
```diff
@@ -0,0 +1,61 @@
+
+const struct option long_options[] =
+{
+   {"version", no_argument, (int *)NULL, 'V'},
```
Done. You are right no cast needed anymore.

sebhz commented on src/argparse.cpp:
```diff
@@ -0,0 +1,61 @@
+      switch (c)
+      {
+         case 'V':
+            std::cout << VERSION_STRING << "\n";
```
Done. I'd rather not add the URL in the --version switch, as this is mostly intended to query the version. But I added it the the --help switch. And you are right, I simply forgot to exit there.

sebhz commented on src/cap32.cpp:
```diff
@@ -3189,17 +3188,17 @@ void doCleanUp (void)
-   for (int i = 1; i < argc; i++) { // loop for all command line arguments
-      LOG_DEBUG("Handling arg " << argv[i]);
-      std::string fullpath = stringutils::trim(argv[i], '"'); // remove quotes if arguments quoted
+   for (unsigned int i = 0; i < slot_list.size(); i++) { // loop for all command line arguments
```
Done. I almost never used C++ before, so did not know this form of iteration !

sebhz commented:
I think I addressed all the points above except the not mandatory one - which I agree would require some heavier process. I'll close this pull request and reopen another cleaner one - since I am not 100% sure how to update the pull request.

ColinPitrat commented:
Normally, simply committing and pushing on your fork should update the pull-request.
